src/components/course-calendar/plan.cjsx

```diff
@@ -120,15 +130,22 @@ CoursePlan = React.createClass
     @setState(publishState)
 
   componentWillMount: ->
-    @subscribeToPublishing(@props.item.plan)
-    @syncRoute()
+    {plan} = @props.item
+    publishState = PlanHelper.subscribeToPublishing(plan, @checkPublishingStatus)
+    isViewingStats = @_doesPlanMatchesRoute()
+
+    state = _.extend({isViewingStats}, publishState)
+    @setState(state)
+
     location = @context.router.getLocation()
     location.addChangeListener(@syncRoute)
 
   componentWillReceiveProps: (nextProps) ->
     if @props.item.plan.id isnt nextProps.item.plan.id
-      @subscribeToPublishing(nextProps.item.plan)
+      publishState = PlanHelper.subscribeToPublishing(nextProps.item.plan, @checkPublishingStatus)
       @stopCheckingPlan(@props.item.plan)
+      state = _.extend(@getStateByProps(nextProps), publishState)
```

Comment:
ah, yep that makes sense.  I hadn't factored that line in my internal logic.   I still think it'd be a good idea to flip the `subscribe` and `stopChecking` calls though, unless they're in that order for a reason.  That way 1) other's won't make the same mistake I just did, and 2) it'll be a bit more future proof if the `plan.id` check is ever removed.